src/core/middleware/store.ts

```diff
@@ -0,0 +1,45 @@
+import { destroy, invalidator, create } from '../vdom';
+import Store, { StatePaths, Path } from '../../stores/Store';
+import { Process } from '../../stores/process';
+
+const factory = create({ destroy, invalidator });
+
+export const createStoreMiddleware = <S = any>(initial?: Function) => {
+	let store = new Store();
```

Comment:
The store is factory'd for the middleware, does not use the store from the injector - We might want to support the injector from the store for backwards compatibility (so that users can progressively adopt functional widgets and middleware)